File: math-and-number-theory/twin_prime_checker.cpp
#include <iostream>
#include <cmath>

using namespace std;

bool isPrime(int N){
	if (N < 2) return false;
    for (int i = 2; i <= sqrt(N); i++){
        if (N % i == 0) return false;
    }
    return true;
}

bool isTwinPrime(int N){
	if (!isPrime(N)) return false;
    return isPrime(N - 2) || isPrime(N + 2);
}

int main(){
	int N;
	cin >> N;
	for (int i = 0; i < N; ++i){
		int p;
		cin >> p;
		if (isTwinPrime(p)){
			cout << "true" << endl;
		}
		else{
			cout << "false" << endl;
		}
	}
	return 0;
}

File: math-and-number-theory/twin_prime_checker.py
def is_prime(x):
    if x<=1:
        return False
    for i in range(2,x):
        if x%i==0:
            return False
    return True



def is_twin_prime(x):	
    if is_prime(x) and (is_prime(x-2) or is_prime(x+2)):
        return True
    else:
        return False
	


N = int(input())
for i in range(N):
    p = int(input())
    if is_twin_prime(p):
        print("true")
    else:
        print("false")

